src/ssl.c

```diff
@@ -3142,6 +3142,23 @@ int wolfSSL_inject(WOLFSSL* ssl, const void* data, int sz)
     return WOLFSSL_SUCCESS;
 }
 
+
+int wolfSSL_write_ex(WOLFSSL* ssl, const void* data, int sz, size_t* wr)
+{
+    int ret;
+
+    ret = wolfSSL_write(ssl, data, sz);
+    if (ret >= 0 && wr != NULL) {
+        *wr = (size_t)ret;
+        ret = 1;
+    }
+    else {
+        ret = 0;
+    }
+    return ret;
+}
```

Comment:
> SSL_write_ex() will return 1 for success or 0 for failure. Success means that all requested application data bytes have been written to the SSL connection or, if SSL_MODE_ENABLE_PARTIAL_WRITE is in use, at least 1 application data byte has been written to the SSL connection. Failure means that not all the requested bytes have been written yet (if SSL_MODE_ENABLE_PARTIAL_WRITE is not in use) or no bytes could be written to the SSL connection (if SSL_MODE_ENABLE_PARTIAL_WRITE is in use).
https://docs.openssl.org/3.0/man3/SSL_write/#warnings

I don't see these partial writes being handled here. This might become a very annoying edge case to debug down the road.